src/ui/public/directives/paginate.js

```diff
@@ -40,6 +40,12 @@ uiModules.get('kibana')
     },
     controllerAs: 'paginate',
     controller: function ($scope, $document) {
+      // Unfortunately this controller relies on inherited scope. Lets not fail if onPageChanged happens to not be
+      // implemented.
+      if (!$scope.onPageChanged) {
```

Comment:
What made you choose to set `$scope.onPageChanged` to a noop function, as opposed to checking to see if it exists before invoking it? Since we're using inherited scope, couldn't us defining it here shadow it's declaration in the "parent" scope if it was set after us setting it here to noop?